optimum/exporters/tasks.py

```diff
@@ -1798,35 +1852,80 @@ def infer_library_from_model(
 
         Args:
-            model_name_or_path (`Union[str, Path]`):
-                Can be either the model id of a model repo on the Hugging Face Hub, or a path to a local directory
-                containing a model.
-            model (`Union[PreTrainedModel, TFPreTrainedModel]`):
-                The instance of the model.
+            model (`Union[str, PreTrainedModel, TFPreTrainedModel, DiffusionPipeline, Type]`):
+                The model to infer the task from. This can either be the name of a repo on the HuggingFace Hub, an
+                instance of a model, or a model class.
             subfolder (`str`, defaults to `""`):
                 In case the model files are located inside a subfolder of the model directory / repo on the Hugging
                 Face Hub, you can specify the subfolder name here.
             revision (`Optional[str]`, *optional*, defaults to `None`):
                 Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id.
             cache_dir (`Optional[str]`, *optional*):
                 Path to a directory in which a downloaded pretrained model weights have been cached if the standard cache should not be used.
-            library_name (`Optional[str]`, *optional*)::
-                The library name of the model. Can be any of "transformers", "timm", "diffusers", "sentence_transformers".
+            token (`Optional[Union[bool,str]]`, defaults to `None`):
+                The token to use as HTTP bearer authorization for remote files. If `True`, will use the token generated
+                when running `huggingface-cli login` (stored in `huggingface_hub.constants.HF_TOKEN_PATH`).
+
+        Returns:
+            `str`: The library name automatically detected from the model repo, model instance, or model class.
+        """
+
+        if isinstance(model, str):
+            library_name = cls._infer_library_from_model_name_or_path(
+                model, subfolder=subfolder, revision=revision, cache_dir=cache_dir, token=token
+            )
+        elif type(model) == type:
+            library_name = cls._infer_library_from_model_or_model_class(model_class=model)
+        else:
+            library_name = cls._infer_library_from_model_or_model_class(model=model)
+
+        return library_name
+
+    @classmethod
+    def standardize_model_attributes(cls, model: Union["PreTrainedModel", "TFPreTrainedModel", "DiffusionPipeline"]):
+        """
+        Updates the model for export. This function is suitable to make required changes to the models from different
+        libraries to follow transformers style.
+
+        Args:
+            model (`Union[PreTrainedModel, TFPreTrainedModel, DiffusionPipeline]`):
+                The instance of the model.
+
         """
-        library_name = TasksManager._infer_library_from_model(model, library_name)
+
+        library_name = TasksManager.infer_library_from_model(model)
```

Comment:
user might specify the library name (for example whether to load the model with `sentence-transformers` or `transformers`)  in which case there could be a mismatch with the inferred `library_name`

Reply:
actually here the model should already be loaded so shouldn't be an issue